packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -243,6 +270,7 @@ class ThemeData with Diagnosticable {
     AndroidOverscrollIndicator? androidOverscrollIndicator,
     bool? applyElevationOverlayColor,
     NoDefaultCupertinoThemeData? cupertinoOverrideTheme,
+    Map<Object, ThemeExtension<Object>>? extensions,
```

Comment:
Would it make sense here and other places to make 
`Iterable<ThemeExtension<Object>>` extensions instead?
As storage/deduplication logic probably should be hidden from ThemeData users?
As well there is no guarante that incorect map would be passed (`{AExtension: BExtension()}`).

And then later:
`extensions: { for(final extension in extensions) extension.id: extension },`

Reply:
I like that idea

Reply:
You're correct, the `const ThemeData.raw()` constructor requires const valued parameters so we can't construct a Map() from a `Iterable<ThemeExtension<Object>>`. In addition to `ThemeData.lerp()` apps are likely to lookup their theme extension[s] frequently, which means that a Map would be a better choice.

Reply:
Would it make sense for raw have Map<Object, ThemeExtension<Object> and Iterable<ThemeExtension<Object>> for factory and copyWith?

Reply:
It deviates from what we've done for `ThemeData` but the API is much cleaner so I've incorporated the suggestion